src/curl.c

```diff
@@ -95,7 +95,7 @@ int swupd_curl_init(void)
 		i = 1;
 
 		tok = strtok_r(str, ":", &ctx);
-		do {
+		while (tok) {
 			if (stat(tok, &st)) {
```

Comment:
Don't you just hate infinite loops?

Consider what happens if stat fails!

Please can we fix this code so it is idiomatic, i.e. replace 98 with

    for ( tok = strtok_r(str, ":", &ctx); tok ; tock = strtok_r(NULL, ":", &ctx) {

and remove 97 and 111

Reply:
Thanks for catching this :-)  Tested with valid paths but without invalid ones.

Done.